tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ProjectEncodingTest.java

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Ingo Mohr - Issue #166 - Add Preference to Turn Off Warning-Check for Project Specific Encoding
+ *******************************************************************************/
+package org.eclipse.core.tests.resources;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.eclipse.core.internal.resources.PreferenceInitializer;
+import org.eclipse.core.internal.resources.ValidateProjectEncoding;
+import org.eclipse.core.internal.utils.Messages;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.osgi.util.NLS;
+import org.hamcrest.*;
+import org.junit.Test;
+
+/**
+ * Test for integration of marker
+ * {@link org.eclipse.core.resources.ResourcesPlugin#PREF_MISSING_ENCODING_MARKER_SEVERITY}.
+ */
+public class ProjectEncodingTest extends ResourceTest {
+
+	private static final int IGNORE = -1;
+
+	private IProject project;
+
+	@Override
+	protected void tearDown() throws Exception {
+		if (project != null) {
+			project.delete(true, null);
+		}
+		InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).putInt(
+				ResourcesPlugin.PREF_MISSING_ENCODING_MARKER_SEVERITY,
+				PreferenceInitializer.PREF_MISSING_ENCODING_MARKER_SEVERITY_DEFAULT);
+		InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).flush();
```

Comment:
Don't think this is necessary.

Reply:
I took-over this like from class ``NatureTest`` (the approach to implement this like for the unknown-nature-check was suggested when I started to work on this issue).

I'd like to keep it as it's a safeguard for tests that run later on because it forces to write the default changes back to the store. (if I'm getting the jdoc right).